parcels/codegenerator.py

```diff
@@ -164,6 +164,8 @@ def __init__(self, grid, ptype):
         self.grid = grid
         self.ptype = ptype
         self.field_args = OrderedDict()
+        # Hack alert: JIT requires U field to update grid indexes
+        self.field_args['U'] = grid.U
```

Comment:
Thanks @mlange05, the PR looks all good. Only thing I don't complete understand is why we only do this line 168 in `codegenerator.py` for `grid.U` and not for the other `grid` fields. Is this because it's a hack?

Anyway, I'm happy with you pulling this one into master